src/stringutils.cpp

```diff
@@ -86,6 +87,26 @@ bool endsWith(const std::string& str, const std::string& match) {
            });
 }
 
+bool startsWith(const char* str, const char* prefix) {
+    // Ensure null terminated
+    const int MAX = 300;
+    const char* end = str;
+    for (; *end != '\0'; ++end) {
+        if ((end - str) > MAX) {
+            return false;
+        }
+    }
+    const char* end2 = prefix;
+    for (; *end2 != '\0'; ++end2) {
+        if ((end2 - str) > MAX) {
+            return false;
+        }
+    }
+    size_t strLen = std::strlen(str);
```

Comment:
Why can't we just use here std::string? 

Then you can compare the sizes first (if prefix len <= checked string len) and then use memcmp with prefix len?